src/routes/docs/tutorials/nextjs/step-5/+page.markdoc

```diff
@@ -0,0 +1,283 @@
+          setPopup={setPopup} 
+          createExpense={createExpense} 
+        />
+      )}
+      <div className="container u-flex-vertical u-gap-12 u-cross-center">
+        <h1 className="heading-level-1 font-bold">Your expenses</h1>
+        <button
+          className="button"
+          onClick={() => setPopup({ type: "new", data: null })}
+        >
+          <span className="icon-plus" aria-hidden="true"></span>
+          <span className="text">New Expense</span>
+        </button>
+      </div>
+    </>
+  );
+};
+
+export default Home;
+```
+
+We are rendering a new component `CreateExpensePopup` when the user clicks on the `New Expense` button. The `CreateExpensePopup` component will be used to create a new expense. 
+
+Let's create the `CreateExpensePopup` component. Create a new file `/src/components/CreateExpensePopup.js` and add the following code.
+
+```js
+"use client";
+
+import React from "react";
+import toast from "react-hot-toast";
+
+const CreateExpensePopup = ({ setPopup, createExpense }) => {
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const form = e.target;
+    const formData = new FormData(form);
+
+    const item = {
+      title: formData.get("title"),
+      amount: formData.get("amount"),
+      type: formData.get("type"),
+    };
+
+    if (!item.title) {
+      return toast.error("Title is required");
+    }
+    if (!item.amount) {
+      return toast.error("Amount is required");
+    }
+    if (!item.type) {
+      return toast.error("Type is required");
+    }
+
+    try {
+      await createExpense(item);
+    } catch (error) {
+      toast.error(error.message);
+    }
+  };
+
+  return (
+    <div className="u-position-fixed u-inset-0 u-width-full-line u-full-screen-height u-flex u-cross-center u-main-center u-z-index-10 lite-bg">
+      <div className="card u-max-width-600 u-width-full-line">
+        <button
+          onClick={() => setPopup(null)}
+          className="u-position-absolute u-inset-inline-end-16 u-inset-block-start-16"
+        >
+          <span className="icon-x-circle" />
+        </button>
+        <p className="heading-level-3 font-bold u-text-center">
+          Create Expense
+        </p>
+        <form className="u-margin-block-start-12" onSubmit={handleSubmit}>
+          <div>
+            <label className="label">Title</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Title"
+                name="title"
+                type="text"
+                required
+              />
+            </div>
+          </div>
+          <div className="u-margin-block-start-12">
+            <label className="label">Amount($)</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Title"
+                name="amount"
+                type="number"
```

Comment:
<img width="587" alt="Screenshot 2023-10-24 at 2 39 20 PM" src="https://github.com/appwrite/website/assets/29069505/f8e5c6d0-cecf-4117-a8ba-d6332fcced3b">
I think this is only taking ints